authorisation-adjustment-example/src/main/java/com/adyen/checkout/api/ApiController.java

```diff
@@ -0,0 +1,189 @@
+
+        log.info("REST request to get Adyen payment methods {}", paymentMethodsRequest);
+        var response = paymentsApi.paymentMethods(paymentMethodsRequest);
+        return ResponseEntity.ok()
+            .body(response);
+    }
+
+    /**
+     * {@code POST  /initiatePayment} : Make a payment.
+     *
+     * @return the {@link ResponseEntity} with status {@code 200 (Ok)} and with body the paymentMethods response.
+     * @throws IOException  from Adyen API.
+     * @throws ApiException from Adyen API.
+     */
+    @PostMapping("/pre-authorisation")
+    public ResponseEntity<PaymentResponse> payments(@RequestHeader String host, @RequestBody PaymentRequest body, HttpServletRequest request) throws IOException, ApiException {
+        var paymentRequest = new PaymentRequest();
+
+        var orderRef = UUID.randomUUID().toString();
+        var amount = new Amount()
+            .currency("EUR")
+            .value(24999L); // value is 249.99€ in minor units
+
+        paymentRequest.setMerchantAccount(this.applicationProperty.getMerchantAccount()); // required
+        paymentRequest.setChannel(PaymentRequest.ChannelEnum.WEB);
+        paymentRequest.setReference(orderRef); // required
+        paymentRequest.setReturnUrl(request.getScheme() + "://" + host + "/api/handleShopperRedirect?orderRef=" + orderRef);
+
+        paymentRequest.setAmount(amount);
+        var authenticationData = new AuthenticationData();
+        authenticationData.setAttemptAuthentication(AuthenticationData.AttemptAuthenticationEnum.ALWAYS);
+        paymentRequest.setAuthenticationData(authenticationData);
+
+        paymentRequest.setAdditionalData(new HashMap<>() { {
+            put ("authorisationType", "PreAuth");
+        }});
+
+        // required for 3ds2 redirect flow
+        paymentRequest.setOrigin(request.getScheme() + "://" + host);
+        // required for 3ds2
+        paymentRequest.setBrowserInfo(body.getBrowserInfo());
+        // required by some issuers for 3ds2
+        paymentRequest.setShopperIP(request.getRemoteAddr());
+        paymentRequest.setPaymentMethod(body.getPaymentMethod());
+
+        // we strongly recommend that you the billingAddress in your request
+        // card schemes require this for channel web, iOS, and Android implementations
+        //paymentRequest.setBillingAddress(new Address());
+        log.info("REST request to make Adyen payment {}", paymentRequest);
+        var response = paymentsApi.payments(paymentRequest);
+
+        if (response.getResultCode() == PaymentResponse.ResultCodeEnum.AUTHORISED) {
+            var payment = new PaymentModel(response.getMerchantReference(),
+                    response.getPspReference(),
+                    response.getAmount().getValue(),
+                    response.getAmount().getCurrency(),
+                    LocalDateTime.now(),
+                    LocalDateTime.now().plusDays(28),
```

Comment:
Could a comment here like in Node to explain why we add 28 days to calculate the date

```
    // add 28 days for define pre-authorisation expiry date
    // The value of '28' varies per scheme, see https://docs.adyen.com/online-payments/adjust-authorisation/#validity.
```